xbmc/guilib/GUIWindow.cpp

```diff
@@ -145,21 +143,22 @@ bool CGUIWindow::LoadXML(const std::string &strPath, const std::string &strLower
     }
 
     // store XML for further processing if window's load type is LOAD_EVERY_TIME or a reload is needed
-    m_windowXMLRootElement = static_cast<TiXmlElement*>(xmlDoc.RootElement()->Clone());
+    m_windowXMLRootElement.reset(static_cast<TiXmlElement*>(xmlDoc.RootElement()->Clone()));
   }
   else
     CLog::Log(LOGDEBUG, "Using already stored xml root node for {}", strPath);
 
   return Load(Prepare(m_windowXMLRootElement).get());
 }
 
-std::unique_ptr<TiXmlElement> CGUIWindow::Prepare(TiXmlElement *pRootElement)
+std::unique_ptr<TiXmlElement> CGUIWindow::Prepare(const std::unique_ptr<TiXmlElement>& rootElement)
 {
-  if (!pRootElement)
+  if (!rootElement)
     return nullptr;
 
   // clone the root element as we will manipulate it
-  auto preparedRoot = std::unique_ptr<TiXmlElement>(static_cast<TiXmlElement*>(pRootElement->Clone()));
+  auto preparedRoot =
+      std::make_unique<TiXmlElement>(*static_cast<TiXmlElement*>(rootElement->Clone()));
```

Comment:
why call Clone if TiXmlElement has a copy constructor?
```c++
auto preparedRoot = std::make_unique<TiXmlElement>(*RootElement))
```
should work and avoid the leak.